space-concordia-robotics/robotics-prototype
# Cxx unit testing 471

# Assignee Section
 ## Description 
Unit testing for C++ code. This PR will include a basic example test directory in science just for testing purposes. Check the wiki 
 (/wiki/CXX-----Unit-Testing) for a detailed explanation on how everything works. 

### Steps for Testing 
1. `mkdir build && cd build` (in robot/rover) 
2. `cmake ..` (Make sure the installation of AUnit and UnixHostDuino are logged) 
3. `make` (We are looking for the 'ScienceTests' target to be built) 
4. `cd ../science/tests` 
5. `make`
6. `./ScienceTests.out` (Check that tests run)
7. `make clean` 

closes #471 

# Reviewer Section Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process. 
- [ ] [General Integration Test (https://docs.google.com/document/d/1ug0CpA1cIzURP8DDFSvCt2CEJJSwJ6Ta6B1LG_hYk6I/edit)Performed Successfully For Pull Requests that do not include code changes, it is not required to perform the tests above.
